src/PlayerDeath.java

```diff
@@ -15,15 +18,30 @@
 
 	@EventHandler(priority = EventPriority.NORMAL,ignoreCancelled=true)
 	public void onPlayerDeath(PlayerDeathEvent event) {
-
-		if (Event.isPlayerActive(event.getEntity().getUniqueId()))
+		
+		/** If player is in the event and if keepInv is true
+		 * then item drops are disabled and their inventory is saved
+		 * for later retrieval
+		 */
+		
+		boolean keepInv = Event.getKeepMyInventory();
+		if (Event.isPlayerActive(event.getEntity().getUniqueId())) {
 			event.setDroppedExp(0);
-
+		
+ 		if (keepInv == true) {
+ 			ItemStack[] content = event.getEntity().getInventory().getContents();
+ 			PlayerRespawn.items.put(event.getEntity(), content);
+ 			event.getEntity().getInventory().clear();
```

Comment:
`event.setKeepInventory(true);`?

(Also don't think you need the separate `keepInv` variable, just `if (Event.getKeepMyInventory)` would be cleaner.)

Reply:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/PlayerDeathEvent.html#setKeepInventory(boolean)

What I mean by this is you should be able to just use `event.setKeepInventory(true);` instead of the three lines
```
 			ItemStack[] content = event.getEntity().getInventory().getContents();
 			PlayerRespawn.items.put(event.getEntity(), content);
 			event.getEntity().getInventory().clear();
```
And then you should also be able to completely skip all the changes to PlayerRespawn.java (probably you'll also need to check if the player died using /quitevent though.)